query/query_manager.rs

```diff
@@ -72,6 +82,10 @@ impl QueryManager {
         let arced_premable = Arc::new(translated_preamble);
         let arced_stages = Arc::new(translated_stages);
         let arced_fetch = Arc::new(translated_fetch);
+        match validate_no_cycles(&arced_premable.iter().enumerate().map(|(i, translated)| (i, translated)).collect()) {
+            Ok(_) => {}
+            Err(typedb_source) => return Err(QueryError::FunctionRetrieval { typedb_source }),
```

Comment:
No, we can generalise this error now?

Reply:
The name of it at least

Reply:
Updated. Please check & see if we need to box the error. I don't know what the current strategy is. Box everywhere?

Reply:
Right now we only box the results that have Large errors in them - if Clippy doesn't warn us on that we don't need to atm.